### PR TITLE
Implement basic TreeView provider for TFC runs

### PR DESCRIPTION
**Pull request by radeksimko**
Implements parts of https://github.com/hashicorp/vscode-terraform/issues/1395

## UX

Screenshot 2023-05-26 at 14 48 30

### CODE REVIEW

File: src/providers/tfc/runProvider.ts
Comment (jpogran) [inline src/providers/tfc/runProvider.ts:31]: We don't need as much info here as you're collecting. Each run has a link attribute with a link to itself https://developer.hashicorp.com/terraform/cloud-docs/api-docs/run#list-runs-in-a-workspace

Comment (radeksimko) [reply src/providers/tfc/runProvider.ts:35]: I also don't like the idea of building out the URL here, but AFAICT that self link is unfortunately just a link to the API, not the "browser" version.

```json
"links": {
    "self": "/api/v2/runs/run-bWSq4YeYpfrW4mx7"
}
```

File: package.json
Comment (jpogran) [inline package.json:471]: Command names have to be unique across the entirety of VS Code, so to prevent confusion and collision I would like to name this something more descriptive than `view`. We have the word `view` everywhere so it's hard to determine what this command will do just by reading the name.

How about `terraform.cloud.run.viewInBrowser`?

The title can remain the same as from the user POV they see a globe and view and that implies internet.

File: src/providers/tfc/runProvider.ts
Comment (jpogran) [inline src/providers/tfc/runProvider.ts:132]:
```suggestion
### \`${workspaceName}\`
#### ID: \`${id}\`
___
${attributes.message}
___
| | |
--|--
| **Status**               | ${attributes.status}|
| **Trigger reason** | ${attributes['trigger-reason']}|
| **Created at**        | ${attributes['created-at']}|
| **Source**              | ${attributes.source}|
| **Terraform**         | ${attributes['terraform-version']}|
```

Love what is there already. This suggestion brings it a little closer to what the TFC page looks like while also spacing things automatically

File: src/providers/tfc/runProvider.ts
Comment (jpogran) [inline src/providers/tfc/runProvider.ts:171]: We can style these with color to indicate success or failure, but that can be a followup

**Review by jpogran — CHANGES_REQUESTED** [review]
Minor suggestions to improve display

File: src/providers/tfc/runProvider.ts
Comment (radeksimko) [reply src/providers/tfc/runProvider.ts:174]: I'm happy to do that but I am really struggling to understand what the best way to do it with the existing VS Code APIs https://code.visualstudio.com/api/references/theme-color

What meanings/IDs should I choose for each occasion? I don't see any appropriate "success" or "fail" ones in the long list. Or did you imply we just declare our own colours? How would I go about that? Implement custom `vscode.ThemeColor` which accepts the colour in hex format or something like that?

How do we know that will render well in all light and dark themes?

I'm leaning more on leaving it for later.

**Comment by radeksimko** [comment]
I really wanted to print human-readable versions of the status, trigger reason and the source but I'm not even sure whether that's what we want to be displaying and I was struggling to find what each key even means (even through browsing the internal atlas codebase).

Similarly, I wanted to print the dates in "time ago" format using https://www.npmjs.com/package/javascript-time-ago but I kinda struggled with that library, specifically with setting up the language.

We could render some additional details in the hover tooltip, like branch name, SHA and author + avatar but all of that data requires an additional request to https://developer.hashicorp.com/terraform/cloud-docs/api-docs/configuration-versions#show-a-configuration-version-s-commit-information which complicates things quite a bit.

I also omitted pagination, in part because we haven't decided what the best way of implementing that would be.

--- 

I'm open to discuss any of the above further.

**Review by jpogran — APPROVED** [review]
LGTM as is! We can iterate more after we land org login and workspace view

**Pull request merged**
